particle/particle/particle.py

```diff
@@ -564,27 +564,44 @@ def from_dec(cls, name):
         # Others are difficult to match with the standard regex rules.
         # The required mapping is here provided:
         dec_to_pdg_mapping = {
+            'omega(2S)': 'omega(1650)',
             'phi' : 'phi(1020)',
+            'K_L0': 'K(L)0',
+            'K_S0': 'K(S)0',
+            'B_s10': 'B(s1)(5830)0',
+            'anti-B_s10': 'B(s1)(5830)~0',
             'J/psi': 'J/psi(1S)',
             'Upsilon': 'Upsilon(1S)',
+            'Upsilon_2(1D)': 'Upsilon(2)(1D)',
             'Upsilon(5S)' : 'Upsilon(10860)',
             'n0': 'n',
             'p+': 'p',
-            'X_1(3872)': 'X(3872)'
+            'X_1(3872)': 'X(3872)',
+            'Omega_c*0': 'Omega(c)(2770)'
         }
-        if name in dec_to_pdg_mapping.keys():
+        if name in sorted(dec_to_pdg_mapping.keys(), reverse=True):
```

Comment:
Why is this sorting? The `in` operator doesn't care, and the time to sort is high compared to checking membership. And once it's sorted, the `in` operator now has to check membership in a list, which is slower than checking membership in an unordered structure like a set (Python < 3.5).

Reply:
Oh, I meant to add a comment next to the code, actually. These sorts are needed to make sure certain replacements are not done twice for names such as "X" and "anti-X".
Agreed that the sorting should be done outside.